File: python/searches_historic.py
import logging
from datetime import datetime, timedelta

import requests

HOSTNAME = "api.sanctions.io"
BEARER_TOKEN = "ded11a1cbd164242b6bb28c51f1dad5f"
API_VERSION = "2.0"

logging.basicConfig()
logger = logging.getLogger()
logger.setLevel(logging.INFO)


def invoke_searches_historic():
    """
    Example showing how to call the /searches/historic endpoint.
    :return: dict
    """
    url = f"https://{HOSTNAME}/searches/historic"
    headers = {
        "Authorization": f"Bearer {BEARER_TOKEN}",
        "Accept": f"application/json; version={API_VERSION}",
    }

    # we want to retrieve all previous requests from the last 30 days
    from_datetime = datetime.now() - timedelta(days=30)
    from_datetime_tz_aware = from_datetime.astimezone()
    payload = {
        "timestamp": from_datetime_tz_aware.isoformat(),
        "result_count": 1,
    }

    response = requests.get(url, headers=headers, params=payload)
    return response.json()


if __name__ == "__main__":
    json_data = invoke_searches_historic()
    logger.info(f"Found {json_data.get('count', 0)} results.")

File: node/searchesHistoric.js
const axios = require("axios")
const qs = require("querystring")

const hostname = "api.sanctions.io"
const bearerToken = "12345678-12AF-12aF-12aF-1234567890aF"
const apiVersion = "2.1"

let uri = `https://${hostname}/searches/historic`
axios
  .get(uri, {
    headers: {
      Authorization: `Bearer ${bearerToken}`,
      Accept: `application/json; version=${apiVersion}`,
    },
    params: {
      timestamp: "2021-01-04T15:56:41.210100+01:00",
      result_count: 1,
    },
  })
  .then((response) => {
    let json = response.data
    console.log(`Found ${json.count} results.`)
  })
